### ISSUE TITLE
Old `d3-format` dependency (missing support for ~)

### ISSUE THREAD

**Issue by visnup** [kristw/react-vega#20]
Within the latest build (4.0.0), d3-format@1.2 is installed under d3-scale. This causes usage of a format like `.1~%` (using `~` which is new in 1.3) to fail when used for axis formatting (but I think works fine when applied to text marks).

Works fine with vega@4.2.0 directly. Also, because a dist file is generated for this react component, there's no workaround I can think of from my project using react-vega.

An example failing spec is:
```json
{
  "$schema": "https://vega.github.io/schema/vega/v3.json",
  "data": [
    {
      "name": "data_0",
      "values": [ 1, 2, 3 ]
    }
  ],
  "width": 200,
  "height": 21,
  "axes": [
    {
      "format": ".1~%",
      "orient": "bottom",
      "scale": "x",
      "tickCount": {
        "signal": "ceil(width/40)"
      }
    }
  ],
  "marks": [
    {
      "encode": {
        "update": {
          "fill": {
            "value": "transparent"
          },
          "stroke": {
            "value": "#4c78a8"
          },
          "x": {
            "field": "data",
            "scale": "x"
          },
          "y": {
            "mult": 0.5,
            "signal": "height"
          }
        }
      },
      "from": {
        "data": "data_0"
      },
      "style": [
        "point"
      ],
      "type": "symbol"
    }
  ],
  "scales": [
    {
      "domain": {
        "data": "data_0",
        "field": "data"
      },
      "name": "x",
      "range": [
        0,
        {
          "signal": "width"
        }
      ],
      "type": "linear"
    }
  ]
}
```

**Comment by kristw** [comment]
`react-vega` lists `vega-lib` as `peerDependencies`, which I trace the dependencies and `vega-lib` 4.2 use d3-format 1.3

https://github.com/vega/vega-lib/blob/master/yarn.lock

so I am not sure why you got `d3-format` 1.2. 
Are you using yarn or npm?
This is a dumb ask , but have you try `yarn install` again? or remove `node_modules` entirely and install?

**Comment by visnup** [comment]
I've been using yarn. and yeah, I've tried `yarn install` a couple times along with removing `node_modules` too.

To be clear, I'm pretty sure you or whoever published to npm last also resolves to d3-format@1.2 because the dist file I believe contains d3-format@1.2. Maybe check with `yarn list --pattern d3-format`?

I've also opened #21 that seems to fix it for me locally...

**Comment by visnup** [comment]
Can also check to see this failure by editing `spec1.js` in the demo dir to include a left axis format of `".1~%"`:

```diff
diff --git a/demo/src/vega/spec1.js b/demo/src/vega/spec1.js
index ae8e3ed..3ef594b 100644
--- a/demo/src/vega/spec1.js
+++ b/demo/src/vega/spec1.js
@@ -50,7 +50,7 @@ export default {
 
   "axes": [
     { "orient": "bottom", "scale": "xscale" },
-    { "orient": "left", "scale": "yscale" }
+    { "orient": "left", "scale": "yscale", "format": ".1~%" }
   ],
 
   "marks": [
```

**Comment by kristw** [comment]
Addressed in version 4.0.1. 
Thank you for reporting!
